osu.Game/Screens/Play/HUD/PlayerTeamFlag.cs

```diff
@@ -40,10 +42,19 @@ public PlayerTeamFlag()
         }
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(UserLookupCache userLookupCache)
         {
             if (gameplayState != null)
-                flag.Team = gameplayState.Score.ScoreInfo.User.Team;
+            {
+                if (gameplayState.Score.ScoreInfo.User.Team != null)
```

Comment:
This probably needs some extra checks to prevent trying to look up like the autoplay user or guest user or something online. See

https://github.com/ppy/osu/blob/b3965f0dd09c1a24c5c9106034a10fbb27f6fc26/osu.Game/Online/API/Requests/Responses/APIUser.cs#L22

Also would be nice to be able to constrain this fetch specifically to replay playback, because this implementation can lead to unnecessary lookup attempts if the user is genuinely not on a team, but that's kinda not possible so I'm not gonna push for it (closest thing would be to check `gameplayState.PlayingState == LocalUserPlayingState.NotPlaying` but that feels _reeeeally_ dodgy).

Reply:
The lookups are cached so I'm not too concerned about an extra lookup. Or even really concerned about the system one. The hope would be that the `UserLookupCache` would handle such cases so we don't need to locally, I think.

Reply:
> The hope would be that the `UserLookupCache` would handle such cases

Depends on what you mean by "handle" i guess. If "handle" means "it sees that the ID cannot possibly valid and then aborts the lookup", then it doesn't handle it. If it means "it sends out the online request with an user ID of 0 and then doesn't die when the response doesn't include the user", then I guess it does handle it.

Reply:
Would be the same as a restricted user. As long as it caches the "not found" lookup (seems to be the case, via `CacheNullValues => true`) I'm not overly concerned 😅.